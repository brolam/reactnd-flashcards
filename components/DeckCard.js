import React from 'react'
import { StyleSheet } from 'react-native'
import { View, Text, Platform } from 'react-native'
import { Constants } from 'expo'
import PropTypes from 'prop-types'
import { white, gray } from '../styles/colors'

export default function DeckCard({ title, amountOfCards }) {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>{title}</Text>
      <Text style={styles.amountOfCards}>{amountOfCards}</Text>
    </View>
  )
}

DeckCard.propTypes = {
  title: PropTypes.string.isRequired,
  amountOfCards: PropTypes.number.isRequired,
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    flex: 1,
    backgroundColor: white,
    borderRadius: Platform.OS === 'ios' ? 16 : 2,
    padding: 20,
    marginLeft: 10,
    marginRight: 10,
    marginTop: 17,
    justifyContent: 'center',
    shadowRadius: 3,
    shadowOpacity: 0.8,
    shadowColor: 'rgba(0, 0, 0, 0.24)',
    shadowOffset: {
      width: 0,
      height: 3
    },
  },
  title: { fontSize: 24, textAlign: 'center' },
  amountOfCards: { fontSize: 18, color: gray }
});