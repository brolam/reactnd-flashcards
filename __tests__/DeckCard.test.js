import React from 'react'
import DeckCard from '../components/DeckCard'

test('renders without crashing', () => {
  const deckDummy = { title: 'One Deck', amountOfCards: 10 }
  const wrapper = shallow(<DeckCard {...deckDummy} />);
  expect(wrapper).toMatchSnapshot();
});