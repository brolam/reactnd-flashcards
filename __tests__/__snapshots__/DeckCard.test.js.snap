// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderRadius": 16,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 17,
      "padding": 20,
      "shadowColor": "rgba(0, 0, 0, 0.24)",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.8,
      "shadowRadius": 3,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 24,
        "textAlign": "center",
      }
    }
  >
    One Deck
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#757575",
        "fontSize": 18,
      }
    }
  >
    10
  </Text>
</View>
`;
